browser/ui/webui/settings/brave_account_handler.cc

```diff
@@ -5,29 +5,21 @@
 
 #include "brave/browser/ui/webui/settings/brave_account_handler.h"
 
-#include <string>
+#include <utility>
 
-#include "base/check_op.h"
-#include "base/functional/bind.h"
 #include "components/password_manager/core/browser/ui/weak_check_utility.h"
 
-BraveAccountHandler::BraveAccountHandler() = default;
+namespace brave_account {
+BraveAccountHandler::BraveAccountHandler(
+    mojo::PendingReceiver<mojom::BraveAccountHandler> handler,
+    mojo::PendingRemote<mojom::BraveAccountClient> client)
+    : handler_(this, std::move(handler)), client_(std::move(client)) {}
 
 BraveAccountHandler::~BraveAccountHandler() = default;
 
-void BraveAccountHandler::RegisterMessages() {
-  web_ui()->RegisterMessageCallback(
-      "getPasswordStrength",
-      base::BindRepeating(&BraveAccountHandler::GetPasswordStrength,
-                          base::Unretained(this)));
-}
-
-void BraveAccountHandler::GetPasswordStrength(const base::Value::List& args) {
-  CHECK_EQ(args.size(), 2U);
-  auto* password = args[1].GetIfString();
-  CHECK(password);
-
-  AllowJavascript();
-  ResolveJavascriptCallback(
-      args[0], base::Value(password_manager::GetPasswordStrength(*password)));
+void BraveAccountHandler::GetPasswordStrength(
+    const std::string& password,
+    mojom::BraveAccountHandler::GetPasswordStrengthCallback callback) {
+  std::move(callback).Run(password_manager::GetPasswordStrength(password));
```

Comment:
you could (optionally) move this onto `brave_settings_ui` directly - its the only function on the interface